plugins/catalog-backend/src/modules/codeowners/lib/resolve.ts

```diff
@@ -15,27 +15,28 @@
  */
 
 import * as codeowners from 'codeowners-utils';
-import { CodeOwnersEntry } from 'codeowners-utils';
-import { filter, get, head, pipe, reverse } from 'lodash/fp';
+import parseGitUrl from 'git-url-parse';
 
 const USER_PATTERN = /^@.*/;
 const GROUP_PATTERN = /^@.*\/.*/;
 const EMAIL_PATTERN = /^.*@.*\..*$/;
 
 export function resolveCodeOwner(
   contents: string,
-  pattern = '*',
+  catalogInfoFileUrl: string,
 ): string | undefined {
-  const owners = codeowners.parse(contents);
-
-  return pipe(
-    filter((e: CodeOwnersEntry) => e.pattern === pattern),
-    reverse,
-    head,
-    get('owners'),
-    head,
-    normalizeCodeOwner,
-  )(owners);
+  const codeOwnerEntries = codeowners.parse(contents);
+
+  const { filepath } = parseGitUrl(catalogInfoFileUrl);
+  const match = codeowners.matchFile(filepath, codeOwnerEntries);
```

Comment:
I'm so confused about what the old implementation was doing here 😅 

This is nicer 👍

Reply:
Yeah, probably not the best usages of a functional pipeline of curried functions here 😢.